tools/flakes/detect_flakes.py

```diff
@@ -13,92 +13,251 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Detect new flakes introduced in the last 24h hours with respect to the
-previous six days"""
+"""Detect new flakes and create issues for them"""
 
 from __future__ import absolute_import
 from __future__ import division
 from __future__ import print_function
 
 import datetime
+import json
+import logging
 import os
+import pprint
 import sys
-import logging
-logging.basicConfig(format='%(asctime)s %(message)s')
+import urllib2
+from collections import namedtuple
 
 gcp_utils_dir = os.path.abspath(
     os.path.join(os.path.dirname(__file__), '../gcp/utils'))
 sys.path.append(gcp_utils_dir)
 
 import big_query_utils
 
-def print_table(table):
-    kokoro_base_url = 'https://kokoro.corp.google.com/job/'
-    for k, v in table.items():
-      job_name = v[0]
-      build_id = v[1]
-      ts = int(float(v[2]))
-      # TODO(dgq): timezone handling is wrong. We need to determine the timezone
-      # of the computer running this script.
-      human_ts = datetime.datetime.utcfromtimestamp(ts).strftime('%Y-%m-%d %H:%M:%S PDT')
-      job_path = '{}/{}'.format('/job/'.join(job_name.split('/')), build_id)
-      full_kokoro_url = kokoro_base_url + job_path
-      print("Test: {}, Timestamp: {}, url: {}\n".format(k, human_ts, full_kokoro_url))
+GH_ISSUES_URL = 'https://api.github.com/repos/grpc/grpc/issues'
+KOKORO_BASE_URL = 'https://kokoro2.corp.google.com/job/'
+
+def gh(url, data=None):
+  request = urllib2.Request(url, data=data)
+  assert TOKEN
+  request.add_header('Authorization', 'token {}'.format(TOKEN))
+  if data:
+    request.add_header('Content-type', 'application/json')
+  response = urllib2.urlopen(request)
+  if 200 <= response.getcode() < 300:
+    return json.loads(response.read())
+  else:
+    raise ValueError('Error ({}) accessing {}'.format(
+        response.getcode(), response.geturl()))
+
+
+def create_gh_issue(title, body, labels):
+  data = json.dumps({'title': title,
+                     'body': body,
+                     'labels': labels})
+  response = gh(GH_ISSUES_URL, data)
+  issue_url = response['html_url']
+  print('Issue {} created for {}'.format(issue_url, title))
 
 
-def get_flaky_tests(days_lower_bound, days_upper_bound, limit=None):
-  """ period is one of "WEEK", "DAY", etc.
-  (see https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#date_add). """
+def build_kokoro_url(job_name, build_id):
+  job_path = '{}/{}'.format('/job/'.join(job_name.split('/')), build_id)
+  return KOKORO_BASE_URL + job_path
 
+
+def create_issues(new_flakes):
+  for test_name, results_row in new_flakes.items():
+    poll_strategy, job_name, build_id, timestamp = results_row
+    url = build_kokoro_url(job_name, build_id)
```

Comment:
having a kokoro link in the new issue is not useful as those links are relatively short-lived. You need to include a sponge link, which is permanent.

Reply:
Yes, but there's no way (that I know of) to access that Sponge link easily. Other than parsing the HTML of the kokoro page, and I'd rather not do that.